src/Mvc/Mvc.Core/test/Infrastructure/ActionMethodExecutorTest.cs

```diff
@@ -21,6 +22,7 @@ public void ActionMethodExecutor_ExecutesVoidActions()
         var valueTask = actionMethodExecutor.Execute(mapper, objectMethodExecutor, controller, Array.Empty<object>());
 
         // Assert
+        Assert.Equal("VoidResultExecutor", actionMethodExecutor.GetType().Name);
```

Comment:
These types are internal so a type check would work? i.e.

```C#
Assert.IsType<VoidResultExecutor>(actionMethodExecutor);
```

Reply:
@davidfowl All the executors are private classes. The only internal class is the `ActionMethodExecutor`.  I don't think there is enough value in making these classes internal just for testing purposes.